lib/cc/analyzer/location_description.rb

```diff
@@ -8,23 +10,30 @@ def initialize(source_buffer, location, suffix = "")
       end
 
       def to_s
-        if location["lines"]
-          begin_line = location["lines"]["begin"]
-          end_line = location["lines"]["end"]
-        elsif location["positions"]
-          begin_line = position_to_line(location["positions"]["begin"])
-          end_line = position_to_line(location["positions"]["end"])
-        end
-
-        str = render_lines(begin_line, end_line)
-        str << suffix unless str.blank?
-        str
+        [
+          lines,
+          lines.blank? ? "" : suffix,
+        ].join
       end
 
       private
 
       attr_reader :location, :suffix
 
+      def lines
```

Comment:
This PR is already huge just by adding the _magic comment_, can we move any refactor to their own PR to mitigate risk?

Reply:
The only refactoring that was done was in order to stop modifying string literals - this change is necessary in order to add the magic comment.